ui-v2/src/components/flows/queries.tsx

```diff
@@ -215,14 +216,22 @@ export const deploymentsCountQueryParams = (
 
 export const deleteFlowMutation = (
 	id: string,
-): {
-	mutationFn: MutationFunction<void>;
-} => ({
+	options: MutationOptions = {},
+): MutationOptions => ({
 	mutationFn: async () => {
 		await getQueryService().DELETE("/flows/{id}", {
 			params: { path: { id } },
 		});
 	},
+	onSettled: () => {
+		return queryClient.invalidateQueries({
+			predicate: (query) => {
+				const firstKey = query.queryKey[0];
+				return typeof firstKey === "string" && firstKey.includes("flows");
+			},
+		});
+	},
```

Comment:
Felt this is the easiest way to ensure the cache is updating after taking the delete action. So long as the API is reasonably fast, this is a fine UX when coupled to a loading indicator on the Delete button.

However, for future consideration, there is [Optimistic Update](https://tanstack.com/query/latest/docs/framework/react/guides/optimistic-updates) support from Tanstack React Query. It's a bit more code and a bit more cognitive overhead though, and I'm not convinced the juice is worth the squeeze in this particular scenario. An option to be aware of and consider though.